src/test/java/com/saucelabs/bamboo/sod/action/BuildConfiguratorTest.java

```diff
@@ -129,6 +137,18 @@ public void sshDisabled() throws Exception {
         assertTrue("Sauce tunnels not empty", getTunnelMap().isEmpty());
     }
 
+    @Test
+    public void sshEnabledWithUniqueKey() throws Exception {
+        buildDefinition.getCustomConfiguration().put(SODKeys.SSH_ENABLED_KEY, "true");
+        buildDefinition.getCustomConfiguration().put(SODKeys.SSH_USE_GENERATED_TUNNEL_ID, "true");
+        buildDefinition.getCustomConfiguration().put(SODKeys.ENABLED_KEY, "true");
+        CustomVariableContext customVariableContext = mock(CustomVariableContext.class);
```

Comment:
I thought about that option but its lot of work for its worth, i will keep that in mind though if i can initialize it with new